### ISSUE TITLE
[Multiselect] Add support for inline autocompletion

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react#5146]
## Context

The current autocomplete component has a long history of accessibility issues, a lack of API flexibility, and requires merchants to take extra steps to complete tasks due to its implementation of the "list with manual selection" combobox pattern. Recent contributions to the system have exposed the underlying combobox and listbox components that were initially baked into the autocomplete component in order to support composition of the many different patterns combobox and listbox WCAG specs support.

We are introducing a new searchable multi select combobox pattern to the Admin to begin addressing this issue, starting with the tag cards on resource detail pages. Along with better hierarchy, reduced cognitive load, and reduced steps for managing tags, the new pattern we're implementing in the tag card aims to increase the ease and speed of adding tags by implementing the "list with inline autocomplete" combobox pattern. 

`Listbox`

- [WCAG Design pattern spec](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox)  
  - [Keyboard spec](https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction) 

Combobox

- [WCAG Design pattern spec](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox)
  - [Keyboard spec](https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-6)
- [WCAG inline autocomplete example spec](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.0pattern/combobox-autocomplete-both.html)

[Figma board](https://www.figma.com/file/I96Hwzt2GfSt2OyKBd5cOe/Multiselect-autocomplete?node-id=703%3A108517)

## Requirements

I'll be incrementally adding to this list and converting to sub-issues to break up the work by component so I can reference them in PRs without prematurely closing this sub-epic:

- [x] #5149
  - [ ]  The `aria-autocomplete` attribute should support being set to `both`
  - [ ] Add a `selection` prop
  - [ ] Call `setSelectionRange` on the `inputRef` when `selection` is provided

- [x] #5150
  - [x] Update `aria-activedescendant` logic for initial render (no input value) so that it updates context properly (see #3811)
    - [x] Default to the first option in the listbox if none are selected
    - [x] Otherwise default to first selected option in the listbox

- [x] #5151

**Comment by laurkim** [comment]
[PR](https://github.com/Shopify/polaris-react/pull/5297) for Listbox accessibility updates

**Comment by laurkim** [comment]
[PR](https://github.com/Shopify/polaris-react/pull/5298) for Combobox updates

**Comment by laurkim** [comment]
Resolved by [#5303](https://github.com/Shopify/polaris/pull/5303).
